app/src/main/java/org/metabrainz/mobile/presentation/features/artist/ArtistViewModel.java

```diff
@@ -18,19 +17,11 @@
 public class ArtistViewModel extends LookupViewModel {
 
     private SingleLiveEvent<WikiSummary> artistWiki;
-    private LiveData<Artist> artistData = Transformations.map(repository.initializeData(),
-            data -> new Gson().fromJson(data, Artist.class));
 
     public ArtistViewModel() {
-    }
-
-    public LiveData<Artist> initializeData() {
-        return artistData;
-    }
-
-    @Override
-    public void fetchData() {
-        repository.fetchData("artist", MBID, Constants.LOOKUP_ARTIST_PARAMS);
+        entity = MBEntities.ARTIST;
+        liveData = Transformations.map(repository.initializeData(),
+                data -> new Gson().fromJson(data, Artist.class));
```

Comment:
(I'm reviewing each commit, so maybe this comment is superseeded by other commit later in time…)

I like your refactor. However I'd like to discuss some things to keep learning about architecture :)

I like your idea of one Repository (one source of code) which can hold all entities. However **that implies that the logic of transforming from the abstract “Entity” to the “Artist Entity” is moved to the ViewModel layer** (the presentation layer)

Usually this **presentation layer should not know anything of transforming data**, as data should be given in the proper data type (in this case, the “Artist livedata type”), and this should be done by the domain layer (“someone is asking me data about an artist, therefore I return Artist entity livedata”)

However, as I say, this is just to be thoughtful about these things in the future and to discuss some interesting things now. But I think **we should stay like this and not change anything further**. Why?

- All the layers use `LiveData`. It's part of the Android ecosystem, so if'd like to be 100% clean in the domain layer, we should use something independent of the infrastructure that could be reused across other projects.
- `Transformations` component is part of AndroidX's components. We should not use it in the domain layer as that would couple it to the framework (Android).

(For both cases I'm thinking on `ReactiveX`, which has implementations for each platform and would allow reusing the domain layer on each)


Anyway, this is just to be thoughtful about these things in the future! 🥇

Reply:
Thanks for the insight. I will take a look into ReactiveX. I also believe this is not the current PR can be improved in many ways. But let's take it up further work in this aspect a few weeks later again.